### ISSUE TITLE
Request: Make Background Color Optional

### ISSUE THREAD

**Issue by spyke01** [GuillaumeJasmin/react-table-hoc-fixed-columns#8]
Can we make the background color optional? Perhaps add a separate prop or just rely on the user making this additional change themselves?

**Comment by GuillaumeJasmin** [comment]
Background color of a row ?

**Comment by spyke01** [comment]
Sorry should have explained in more detail, the background color for the fixed elements appears to be hardcoded to white so that the fixed elements don't just overly on the other table column data when scrolling.

Instead of making this white it would be great to have the option of keeping these the default colors and then allowing the developer to specify otherwise if need by via CSS. Right now if you are keeping the default table styling you will have to do the following to get the table to convert back to the default style:

 ```scss
    .analytics .ReactTable .rt-thead, .analytics .ReactTable .rt-tfoot {
      .rt-th:first-child, .rt-td:first-child {
        background-color: #f5f7f8;
      }
    }

   .ReactTable.-striped .rt-tr {
      background: #fff;

      .rt-td:first-child {
        background: #fff !important;
      }
    }

    .ReactTable.-striped .rt-tr.-odd {
     /* Origina BGl is a translucent black overlay so I chose something close */
      background: #f9f9f9;

      .rt-td:first-child {
        background: #f9f9f9 !important;;
      }
    }
```

**Comment by GuillaumeJasmin** [comment]
I added default background color because the default background color is transparent.
So without background, when we start scrolling, we have this:
![screen shot 2018-08-20 at 18 46 57](https://user-images.githubusercontent.com/3513444/44354231-9588d100-a4a9-11e8-8daa-f2b8d614e5a1.png)

So with background colors, we now have this:
![screen shot 2018-08-20 at 18 47 03](https://user-images.githubusercontent.com/3513444/44354293-bb15da80-a4a9-11e8-9ea3-c13d9be23837.png)

So you just want to override it ? It's impossible to have the default bacground color of ReactTable (transparent). But I can add a way to override the white background color ?

**Comment by GuillaumeJasmin** [comment]
Wait, what version do you use ?

**Comment by av-dev** [comment]
@GuillaumeJasmin I would assign a special className to a fixed column and define styles for it (not for all columns in the table). For example, `.fixedColumn {background-color: inherit}`. And I would not use CSSinJS, because it's hard to override it with normal styles (classes can change in newer versions).

**Comment by GuillaumeJasmin** [comment]
@av-dev for all fixed columns (left and right) like this:
```js
<ReactTable
  classNameFixedColumns={}
/>
```

Or maybe 
```js
<ReactTable
  classNameLeftFixedColumns={}
  classNameRightFixedColumns={}
/>
```

Or both ?

**Comment by av-dev** [comment]
@GuillaumeJasmin 
```
<ReactTable
  classNameFixedColumns={}
/>
```
and you can add a modifier `className={`${props.classNameFixedColumns} --${position}`}` (--left or --right). I also meant that I can not change your styles in your project because they can change

**Comment by GuillaumeJasmin** [comment]
Yes styles can change because I use [emotion](https://github.com/emotion-js/emotion).

Do you prefer have props like `classNameFixedColumns `, or a hard coded className that you can be override with CSSinJS ?

**Comment by av-dev** [comment]
@GuillaumeJasmin I think that using CSS-in-JS(same emotion, styled-components) in open-source is not a good idea. And if possible, please remove the superfluous styles from non-fixed columns (background, background on the hover, etc.). It's hard to override

**Comment by GuillaumeJasmin** [comment]
I agree that css in js is not a good thing for open-source project. I will change it.

ReactTable provide `-striped` and `highlight` className for row. However, this doesn't work with this HOC, because fixed columns must have background color (see screenshot in previous comment).

So I added the same behavior, a fallback in order to have `-striped` and `highlight` className still working. So if I remove this background feature, `-striped` and `highlight` can't work

**Comment by GuillaumeJasmin** [comment]
This issue will be fixed with `v2` release, which is currently in beta. [See release note](https://github.com/GuillaumeJasmin/react-table-hoc-fixed-columns/releases/tag/v2.0.0-beta.1)

**Comment by CRGavrila** [comment]
Rthfc needs to be more flexibility when it comes to background of fixed columns, I have dark and light theme and I should change the "white" background.

I have to do something like this:
`
getTdProps={(a, b, { fixed }) => {
                  return ({
                    className: classes.cellView,
                    style: {
                      background: fixed ? theme.palette.background.paper : 'inherit'
                    }
                  });
                }}
`
I couldn't make to work with none of following:
  stripedColor="#f00"
  highlightColor="#f00"
  classNameFixedColumns
